### PR TITLE
nsenter: remove a proxy process

### PR DESCRIPTION
**Pull request by avagin**
Currently nsexec() creates a proxy process to enter into a pid namespace.
It isn't good, because we need to proxy an exit code and signals.
We can use CLONE_PARENT to fork a process with the right parent.

Signed-off-by: Andrey Vagin <avagin@openvz.org>

### CODE REVIEW

**Comment by vmarmol** [comment]
LGTM, clean and simpler

**Comment by avagin** [comment]
@vmarmol Yes, we can.
https://github.com/docker/libcontainer/blob/api/namespaces/execin.go#L30

Thank you for review

File: namespaces/nsenter/nsexec.c [hunk 1]
```diff
@@ -11,11 +12,32 @@
...
+
+/* All arguments should be above stack, because it grows down */
+struct cr_clone_arg {
+        /*
+         * Reserve some space for clone() to locate arguments
+         * and retcode in this place
+         */
+        char stack[4096] __attribute__((aligned (8)));
+        char stack_ptr[0];
+	jmp_buf *env;
```
Comment (mrunalp) [inline namespaces/nsenter/nsexec.c]: Minor: indentation looks off here.

Comment (avagin) [inline namespaces/nsenter/nsexec.c]: Fixed, thanks.

**Comment by crosbymichael** [comment]
LGTM

**Pull request merged**